src/Sentry.Extensions.AI/SentryAISpanEnricher.cs

```diff
@@ -0,0 +1,216 @@
+    /// <param name="messages">a list of <see cref="ChatResponseUpdate"/></param>
+    /// <param name="aiOptions">AI-specific options</param>
+    internal static void EnrichWithStreamingResponses(ISpan span, List<ChatResponseUpdate> messages,
+        SentryAIOptions aiOptions)
+    {
+        var inputTokenCount = 0L;
+        var outputTokenCount = 0L;
+        var finalText = new StringBuilder();
+
+        foreach (var message in messages)
+        {
+            foreach (var content in message.Contents)
+            {
+                if (content is UsageContent { } usage)
+                {
+                    inputTokenCount += usage.Details.InputTokenCount ?? 0;
+                    outputTokenCount += usage.Details.OutputTokenCount ?? 0;
+                }
+            }
+
+            if (message.ModelId is { } modelId)
+            {
+                span.SetData(SentryAIConstants.SpanAttributes.ResponseModel, modelId);
+            }
+
+            if (message.Text is { } responseText)
+            {
+                finalText.Append(responseText);
+            }
+
+            // Only set tool call info in chat spans
+            if (message.FinishReason == ChatFinishReason.ToolCalls
+                && span.Operation == SentryAIConstants.SpanAttributes.ChatOperation)
+            {
+                PopulateToolCallsInfo(message.Contents, span);
+            }
+        }
+
+        if (aiOptions.RecordOutputs)
+        {
+            span.SetData(SentryAIConstants.SpanAttributes.ResponseText, finalText.ToString());
+        }
+
+        span.SetData(SentryAIConstants.SpanAttributes.UsageInputTokens, inputTokenCount);
+        span.SetData(SentryAIConstants.SpanAttributes.UsageOutputTokens, outputTokenCount);
+        span.SetData(SentryAIConstants.SpanAttributes.UsageTotalTokens, inputTokenCount + outputTokenCount);
+    }
+
+    private static void PopulateToolCallsInfo(IList<AIContent> contents, ISpan span)
+    {
+        var functionContents = contents.OfType<FunctionCallContent>().ToArray();
+        if (functionContents.Length > 0)
+        {
+            span.SetData(SentryAIConstants.SpanAttributes.ResponseToolCalls,
+                FormatFunctionCallContent(functionContents));
+        }
+    }
+
+    private static string FormatAvailableTools(IList<AITool> tools)
+    {
+        return FormatAsJsonList(tools, tool => new
+        {
+            name = tool.Name,
+            description = tool.Description
+        });
+    }
+
+    private static string FormatRequestMessage(ChatMessage[] messages)
+    {
+        return FormatAsJsonList(messages, message =>
+        {
+            var content = message.Role == ChatRole.Tool ? FunctionCallToString(message.Contents) : message.Text;
+
+            return new
+            {
+                role = message.Role,
+                content
+            };
+        });
+
+        object FunctionCallToString(IList<AIContent> toolContents)
```

Comment:
```suggestion
        string FunctionCallToString(IList<AIContent> toolContents)
```

This should be a string right?